### PR TITLE
Update lisk-dpos to use encoded data - Closes #5266

### LINKED ISSUE #5266: Update lisk-dpos to use encoded data
### Description

- Define schema for voteWeights and forgersList
- Store encoded value and used decoded value using stateStore
- Update CHAIN_STATE_DELEGATE_USERNAMES to use binary data
- update delegate transaction (registration) based on changes

### Motivation

`lisk-codec` should be used consistently across the `lisk-sdk`

### Acceptance Criteria

- All data stored in dpos has schema defined
- Update unit tests

### Additional Information

Related issues: #5200

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

This PR resolves #5266

### How was it solved?

- By removing usage of JSON and replacing with lisk-codec
- By adding schemas for voteWeights
- By adding schemas for forgersList
- By adding schemas for Delegate User Names
- By updating the delegate registration transaction

### How was it tested?

Unit tests

### CODE REVIEW

File: elements/lisk-dpos/src/delegates_list.ts [hunk 1]
```diff
@@ -308,14 +289,17 @@ export class DelegatesList {
 		);
 
 		let usernames = { registeredDelegates: [] } as ChainStateUsernames;
+
 		if (usernamesBuffer) {
-			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-			const parsedUsername = JSON.parse(usernamesBuffer.toString('utf8'));
+			const parsedUsernames = codec.decode(
```
Comment (shuse2) [inline elements/lisk-dpos/src/delegates_list.ts:294]:
```suggestion
			const parsedUsernames = codec.decode< DecodedUsernames>(
```

File: elements/lisk-dpos/src/delegates_list.ts [hunk 2]
```diff
@@ -308,14 +289,17 @@ export class DelegatesList {
...
+				(delegatesUserNamesSchema as unknown) as Schema,
+				usernamesBuffer,
+			);
+
 			usernames = {
-				// eslint-disable-next-line
-				registeredDelegates: parsedUsername.registeredDelegates.map(
-					(names: { address: string; username: string }) => ({
-						address: Buffer.from(names.address, 'binary'),
+				registeredDelegates: (parsedUsernames as DecodedUsernames).registeredDelegates.map(
```
Comment (shuse2) [inline elements/lisk-dpos/src/delegates_list.ts:300]: with above, u shouldn't need to parse here

File: elements/lisk-dpos/src/delegates_list.ts [hunk 3]
```diff
@@ -494,10 +481,17 @@ export class DelegatesList {
 		if (!forgersListBuffer) {
 			throw new Error(`No delegate list found for round: ${round.toString()}`);
 		}
-		const forgersList = decodeForgerList(forgersListBuffer);
 
-		const delegateAddresses = forgersList.find(fl => fl.round === round)
-			?.delegates;
+		const forgersListDecoded = codec.decode(
```
Comment (shuse2) [inline elements/lisk-dpos/src/delegates_list.ts:485]:
```suggestion
		const forgersListDecoded = codec.decode<ForgersList>(
```
same as above

File: elements/lisk-dpos/src/schemas.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
+export const voteWeightsSchema = {
```
Comment (shuse2) [inline elements/lisk-dpos/src/schemas.ts:1]: please add license header

Comment (pablitovicente) [reply elements/lisk-dpos/src/schemas.ts:15]: done

File: elements/lisk-dpos/src/delegates_list.ts (hunk 1 above)
Comment (pablitovicente) [reply elements/lisk-dpos/src/delegates_list.ts]: Fixed locally to made the fix in your next comment below. Thanks for this suggestion.

File: elements/lisk-dpos/src/delegates_list.ts (hunk 3 above)
Comment (pablitovicente) [reply elements/lisk-dpos/src/delegates_list.ts:485]: As discussed that it fires more changes. We leave as part of fix of types in lisk-codec. Thanks for this suggestion.

**Pull request merged**